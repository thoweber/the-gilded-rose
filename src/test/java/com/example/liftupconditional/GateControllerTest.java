package com.example.liftupconditional;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GateControllerTest {

    @Test
    void unlock() {
        CombinationApprovals.verifyAllCombinations(
                GateController::unlockWith,
                Keycard.values()
        );
    }

}
