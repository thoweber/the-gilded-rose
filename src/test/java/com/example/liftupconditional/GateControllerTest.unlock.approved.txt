[RED_KEYCARD] => [WOODEN_GATE, IVORY_GATE, DARK_GATE] 
[BLUE_KEYCARD] => [COPPER_GATE, IVORY_GATE, STEEL_GATE, LAVA_GATE] 
[YELLOW_KEYCARD] => [DARK_GATE, STEEL_GATE] 
[GREEN_KEYCARD] => [IVORY_GATE, STEEL_GATE, LAVA_GATE] 
